attractions/utils/range.js

```diff
@@ -1,8 +1,8 @@
 /**
  * Generates a semi-open range.
  * @param {number} start The beginning of the range (included)
- * @param {number} end The end of the range (excluded)
- * @param {number} step The distance between the numbers in the range
+ * @param {number} [end] The end of the range (excluded)
```

Comment:
Technically it's `start` that is optional, not `end`

Reply:
You cannot have an optional parameter followed by a non-optional one. Again, we are limited by TypeScript's lack of support for overloading in JSDoc.